pynestml/transformers/assign_implicit_conversion_factors_transformer.py

```diff
@@ -0,0 +1,335 @@
+# -*- coding: utf-8 -*-
+#
+# assign_implicit_conversion_factors_transformer.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Sequence, Union
+
+from pynestml.meta_model.ast_compound_stmt import ASTCompoundStmt
+from pynestml.meta_model.ast_declaration import ASTDeclaration
+from pynestml.meta_model.ast_inline_expression import ASTInlineExpression
+from pynestml.meta_model.ast_node import ASTNode
+from pynestml.meta_model.ast_small_stmt import ASTSmallStmt
+from pynestml.meta_model.ast_stmt import ASTStmt
+from pynestml.symbols.error_type_symbol import ErrorTypeSymbol
+from pynestml.symbols.predefined_types import PredefinedTypes
+from pynestml.symbols.symbol import SymbolKind
+from pynestml.symbols.template_type_symbol import TemplateTypeSymbol
+from pynestml.symbols.variadic_type_symbol import VariadicTypeSymbol
+from pynestml.transformers.transformer import Transformer
+from pynestml.utils.ast_source_location import ASTSourceLocation
+from pynestml.utils.ast_utils import ASTUtils
+from pynestml.utils.logger import LoggingLevel, Logger
+from pynestml.utils.logging_helper import LoggingHelper
+from pynestml.utils.messages import Messages
+from pynestml.utils.type_caster import TypeCaster
+from pynestml.visitors.ast_visitor import ASTVisitor
+
+
+class AssignImplicitConversionFactorsTransformer(Transformer):
+    r"""
+    Assign implicit conversion factors in expressions.
+    """
+
+    def transform(self, models: Union[ASTNode, Sequence[ASTNode]]) -> Union[ASTNode, Sequence[ASTNode]]:
```

Comment:
Which are the two cocos that are replaced by this transformer? Do they also have to be removed from the cocos manager class?

Reply:
Thank you for remarking upon this. The calls  have been effectively removed by setting ``set_implicit_conversion_factor_on_lhs`` to False on https://github.com/nest/nestml/pull/1105/files#diff-e13518a54108e0f87db406aee6c43eadbebc0bb6908943a1f045136bc1c14024R43. This affects ``co_co_illegal_expression``, ``co_co_user_defined_function_correctly_defined`` and ``co_co_function_calls_consistent``.

Reply:
Can these cocos be removed then?